pkg/api/index.go

```diff
@@ -39,7 +39,7 @@ func getProfileNode(c *models.ReqContext) *dtos.NavLink {
 		})
 	}
 
-	if !setting.DisableSignoutMenu {
+	if !setting.DisableSignoutMenu && !c.SignedInWithJWT() {
```

Comment:
Can you explain the logic here?  Seems like show/hide should depend on the config -- not the auth type

Reply:
When signed in with JWT, the "Sign out" button does nothing and only confuses the user. This is because we can't invalidate the token sent from external system in http headers. AFAIK, there's no standard means to do something like this.

Reply:
right -- but that is true for other auth methods (auth proxy etc), and why the setting for `DisableSignoutMenu` exists.  When you want to hide the signout button, can you just set the DisableSignoutMenu setting?

Reply:
Ok I'll update the docs on `disable_signout_menu` setting mentioning that it's not only useful with `auth.proxy` but with `auth.jwt` too.